pkg/storage/storagewrappers/combinedtuplereader.go

```diff
@@ -29,11 +29,19 @@ type CombinedTupleReader struct {
 var _ storage.RelationshipTupleReader = (*CombinedTupleReader)(nil)
 
 // filterTuples filters out the tuples in the provided slice by removing any tuples in the slice
-// that don't match the object and relation provided in the filterKey.
-func filterTuples(tuples []*openfgav1.TupleKey, targetObject, targetRelation string) []*openfgav1.Tuple {
+// that don't match the object, relation or user provided in the filterKey.
+//
+//nolint:unparam
+func filterTuples(tuples []*openfgav1.TupleKey, targetObject, targetRelation string, optionalParams ...string) []*openfgav1.Tuple {
+	var targetUser string
+	if len(optionalParams) > 0 {
+		targetUser = optionalParams[0]
```

Comment:
You could make `targetUser` a slice of users  and then make the comparison `(targetUser.Len() == 0 || slices.Contains(targetUsers,tk.GetUser())`